File: app-flutter-lista-contatos/lib/model/endereco.dart
import 'package:json_annotation/json_annotation.dart';

part 'endereco.g.dart';

@JsonSerializable()
class Endereco {

  const Endereco({ this.cep, this.logradouro, this.complemento, this.bairro, this.localidade });

  factory Endereco.fromJson(Map<String, dynamic> json) => _$EnderecoFromJson(json);

  final String? cep;
  final String? logradouro;
  final String? complemento;
  final String? bairro;
  final String? localidade;

  Endereco toJson() => _$EnderecoFromJson(this as Map<String, dynamic>);

}

File: ConsultaCepTrabalhoFinal/composeApp/src/commonMain/kotlin/org/example/consultaceptrabalhofinal/models/Endereco.kt
package org.example.consultaceptrabalhofinal.models

import kotlinx.serialization.Serializable

@Serializable
data class Endereco(
    val cep: String = "",
    val logradouro: String = "",
    val localidade: String = "",
    val bairro: String = "",
    val uf: String = ""
)